### PR TITLE
continue on enabling ASP.NET Core unit tests

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

* Issue #939, Test case enabled for ASP.NET Core.

### Description

*  Enable ASP.NET Core unit tests for Singleton and DateTime. *

### Checklist (Uncheck if it is not completed)

- [x] *Test cases added*
- [x] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: test/UnitTest/Microsoft.Test.AspNet.OData/Routing/ODataSingletonRoutingTest.cs [hunk 1]
```diff
@@ -74,9 +74,10 @@ public ODataSingletonRoutingTest()
 
             // Act & Assert
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
+#if !NETCORE
```
Comment (robward-ms) [inline test/UnitTest/Microsoft.Test.AspNet.OData/Routing/ODataSingletonRoutingTest.cs]: I'd suggest #if NETFX here.

Comment (xuzhg) [reply test/UnitTest/Microsoft.Test.AspNet.OData/Routing/ODataSingletonRoutingTest.cs]: #resolved.

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore/packages.config [hunk 2]
```diff
@@ -48,10 +48,10 @@
   <package id="Microsoft.Extensions.Primitives" version="2.0.0" targetFramework="net461" />
   <package id="Microsoft.Net.Http.Headers" version="2.0.1" targetFramework="net461" />
   <package id="Microsoft.NETCore.Platforms" version="1.0.1" targetFramework="net461" />
-  <package id="Microsoft.OData.Client" version="7.4.0-beta" targetFramework="net461" />
```
Comment (robward-ms) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore/packages.config]: We should probably do this to all other projects, including perf, as well I think.

Comment (xuzhg) [reply test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNetCore/packages.config]: Yes. I am working on this. See my next PR.

**Pull request merged**